### PR TITLE
Replace hideLower with showLess

### LINKED ISSUE #10158: Replace 'Hide lower' button's label with something else
**Describe the issue you encountered:**
Replace 'Hide lower' button's label with something else. 

https://github.com/brave/browser-laptop/blob/d14fb50a47d09107611a895d0425a4780411c4e9/app/extensions/brave/locales/en-US/preferences.properties#L375.

It is not obvious what `lower` means here (lower than what?). 

- Platform (Win7, 8, 10? macOS? Linux distro?): All

- Brave Version (revision SHA): master branch

- Steps to reproduce:
   1. Run `npm run add-simulated-synopsis-visits 100`
   2. Click show all
    
- Actual result: the label of the button displayed bottom is `Hide Lower`

- Expected result: the label should be replaced with something clearer. eg. Hide small contribution sites, Hide low contribution sites, etc.

- Extra QA steps:
    1.
    2.
    3.

- Any related issues: #9137

### PR DESCRIPTION
**Pull request by luixxiul**
Based on the discussion on Slack.

Also:
- Rename constants to sort them not only alphabetically but semantically (just sorting alphabetically breaks the semantic order of the existing constants)
- Add lacking states to docs/state.md

Closes #10158

Auditors: 

Test Plan:
1. Run `npm run add-simulated-synopsis-visits 100`
2. Open about:preferences#payments
3. Click `Show All` button
4. Click `Show Less` button
5. Make sure the buttons work

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: docs/state.md [hunk 1]
```diff
@@ -163,6 +163,7 @@ AppStore
     'advanced.default-zoom-level': number, // the default zoom level for sites that have no specific setting
     'advanced.hardware-acceleration-enabled': boolean, // false if hardware acceleration should be explicitly disabled
     'advanced.hide-excluded-sites': boolean, // whether to hide excluded sites in the payments list
+    'advanced.show-less-sites': boolean, // whether to show less sites in the payments list
```
Comment (NejcZdovc) [inline docs/state.md]: Please sort this alphabetically

Comment (luixxiul) [inline docs/state.md]: `d -> ha -> hi -> s`, which is not alphabetically?

Comment (luixxiul) [inline docs/state.md]: I thought `hide-excluded-sites` and `show-less-sites` should be grouped.

Comment (NejcZdovc) [inline docs/state.md]: all properties in `state.md` should be sorted alphabetically

Comment (NejcZdovc) [inline docs/state.md]: now is `d -> ha -> hi -> s ->m`

Comment (luixxiul) [inline docs/state.md]: > all properties in state.md should be sorted alphabetically

so should be ones in appConfig.js right? working on it.

File: js/constants/settings.js [hunk 2]
```diff
@@ -52,13 +52,18 @@ const settings = {
   // Autofill
   AUTOFILL_ENABLED: 'privacy.autofill-enabled',
   // Payments Tab
+  PAYMENTS_ALLOW_NON_VERIFIED: 'payments.allow-non-verified-publishers',
```
Comment (luixxiul) [inline js/constants/settings.js]: Renamed based on https://github.com/brave/browser-laptop/pull/10164/files#diff-5367ecbd5cf259d530494fb7eb37fa91L61

File: docs/state.md [hunk 3]
```diff
@@ -192,10 +187,18 @@ AppStore
...
+    'payments.minimum-visit-time': number,
+    'payments.minimum-visits': number,
+    'payments.notification-add-funds-timestamp': number,
+    'payments.notification-reconcile-soon-timestamp': number,
+    'payments.notification-try-payments-dismissed': boolean, // true if you dismiss the message or enable Payments
     'payments.notifications': boolean, // true to show payment notifications
-    'payments.notificationTryPaymentsDismissed': boolean, // true if you dismiss the message or enable Payments
+    'payments.sites-auto-suggest': boolean, // show auto suggestion
+    'payments.sites-hide-excluded': boolean, // whether to hide excluded sites in the payments list
+    'payments.sites-show-less': boolean, // whether to show less sites in the payments list
```
Comment (luixxiul) [inline docs/state.md]: Renamed above 3 ones to group semantically

**Comment by cezaraugusto** [comment]
also moving to 0.21.x

**Comment by luixxiul** [comment]
the milestone was moved: https://github.com/brave/browser-laptop/pull/10164#event-1194629123

**Comment by bsclifton** [comment]
@luixxiul I believe we have a problem here

We can't update the names of the preferences if they are already in production.  People already have an existing value and this value is lost completely when they upgrade

Steps to reproduce:
- Use an existing version of Brave
- dismiss the notification to use Brave payments (this sets `payments.notificationTryPaymentsDismissed`)
- Use this version of Brave (where `payments.notificationTryPaymentsDismissed` was renamed to `payments.notification-try-payments-dismissed`)
- Notice you'll get prompted again (because the setting was lost)

I checked the other values and they all kept the same name (which is good). We have to be careful about changing the names of any existing preferences.  In sessionStore.js, there is a place where you can handle this migration:
https://github.com/brave/browser-laptop/blob/0c575382b6d4ab320c8d015299226b97b932a528/app/sessionStore.js#L465

**Comment by bsclifton** [comment]
The fix would look something like:
```
module.exports.runPreMigrations = (data) => {
  // existing code here...
  if (data.settings['payments.notificationTryPaymentsDismissed'] != null) {
    data.settings['payments.notification-try-payments-dismissed'] = data.settings['payments.notificationTryPaymentsDismissed']
  }
}
```

**Comment by cezaraugusto** [comment]
sorry I was unaware of that and thanks for the heads-up. I'll came along with a follow-up to avoid revert.

**Comment by luixxiul** [comment]
@bsclifton thanks for the heads-up from me too.

@cezaraugusto thanks in advance :-)

**Comment by bsclifton** [comment]
@cezaraugusto you may check out the conversation in https://github.com/brave/browser-laptop/pull/10181#discussion_r131582113 for a proposal that would solve this issue too 😄

**Comment by bsclifton** [comment]
We're running into issues and unfortunately, this PR is causing issues with the BAT Mercury work (since settings were renamed).  I'm going to pull this into 0.19.x

**Comment by bsclifton** [comment]
master https://github.com/brave/browser-laptop/commit/f0d9c3dc13484c318539fee01fcc13a719845864
0.20.x https://github.com/brave/browser-laptop/commit/3bc7c1f5bca72db7cf0db7600bffd50abe953813
0.19.x https://github.com/brave/browser-laptop/commit/be048b8835a681175178c720bb5e203767bb6505

**Pull request merged**